webapp/src/features/adminOrganizationTabUser/adminOrganizationTabUserSlice.js

```diff
@@ -0,0 +1,245 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { selectToken } from "../../generalSlices/userSlice";
+import EnvironmentVars from "../../EnvironmentVars";
+import apiHelper from "../../apis/api-helper";
+
+const initialState = {
+  availableUserList: [],
+  selectedUserList: [],
+  availableRoles: [],
+};
+
+const getUserData = async (email, token) => {
+  return await apiHelper._getDataWithCodes({
+    url: EnvironmentVars.adminUser,
+    token,
+    query_params: { user_email: email },
+    additional_headers: { "Content-Type": "application/json" },
+  });
+};
+
+export const getAvailableRoles = createAsyncThunk(
+  "adminOrganizationTabUser/getAvailableRoles",
+  async (_, { getState }) => {
+    const currentState = getState();
+    const token = selectToken(currentState);
+
+    const data = await apiHelper._getDataWithCodes({
+      url: EnvironmentVars.adminAddUser,
+      token,
+      additional_headers: { "Content-Type": "application/json" },
+    });
+
+    switch (data.status) {
```

Comment:
Should we add a default case statement here?

Reply:
added default case with the same response as 400/500